braindecode/datasets/base.py

```diff
@@ -438,3 +449,114 @@ def set_description(self, description, overwrite=False):
         for key, value in description.items():
             for ds, value_ in zip(self.datasets, value):
                 ds.set_description({key: value_}, overwrite=overwrite)
+
+    def save(self, path, overwrite=False, offset=None):
+        """Save datasets to files by creating one subdirectory for each dataset:
+        path/
+            target_name.json (if target_name is not None and dataset is raw)
+            0/
+                0-raw.fif | 0-epo.fif
+                description.json
+            1/
+                1-raw.fif | 1-epo.fif
+                description.json
+            ...
+
+        Parameters
+        ----------
+        path : str
+            Directory to which -raw.fif | -epo.fif and .json files are stored.
+        overwrite : bool
+            Whether to delete old files (target_name.json, description.json,
+             -raw.fif, -epo.fif) in specified directory prior to saving.
+        offset : int | None
+            If provided the integer is added to the id of the datasets in the
+            concat. This is useful in the setting of very large datasets, where
+            one dataset has to be processed and saved at a time to account for
+            its original position.
```

Comment:
so just for my understanding, this way one can also save into the same parent directory step by step all the datasets? do we have a test for this?

Reply:
This was the idea but it is actually on https://github.com/braindecode/braindecode/pull/277#discussion_r666443800 to implement it.

Reply:
Not a test strictly speaking, but https://github.com/braindecode/braindecode/pull/277#discussion_r666443800 uses it and would crash if it didn't work. Would probably be good to add a dedicated test.